### PR TITLE
Publish dynamic shape doc.

### PR DESCRIPTION
**Pull request by vanbasten23**

### CODE REVIEW

File: docs/dynamic_shape.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+| Compilation cache hit    | 198               | 1953      |
+
+![Performance comparison (a) without dynamic shape  (b) with dynamic shape](assets/dynamic_shape_mlp_perf.png)
+
+One of the motivations of the dynamic shape is to reduce the number of excessive recompilation when the shape keeps changing between iterations. From the figure above, you can see the number of compilations reduced by half which results in the drop of the training time.
+
+To try it out, run
+```
+XLA_EXPERIMENTAL="nonzero:masked_select" PJRT_DEVICE=TPU python3 pytorch/xla/test/ds/test_dynamic_shape_models.py TestDynamicShapeModels.test_backward_pass_with_dynamic_input
+```
```
Comment (miladm) [inline docs/dynamic_shape.md:46]: Feel free to include the following:

"For more details on how we plan to expand the dynamic shape support on PyTorch/XLA in the future, feel free to review our [RFC](https://github.com/pytorch/xla/issues/3884)."

**Review by miladm — APPROVED** [review]
Thanks @vanbasten23! LGTM. Added a small comment.

File: docs/dynamic_shape.md (hunk 1 above)
Comment (vanbasten23) [reply docs/dynamic_shape.md:46]: Thanks. Added it.

**Comment by vanbasten23** [comment]
Thanks for the review.

**Pull request merged**